tests/sentry/api/endpoints/test_api_token_details.py

```diff
@@ -0,0 +1,154 @@
+from rest_framework import status
+
+from sentry.models.apitoken import ApiToken
+from sentry.testutils.cases import APITestCase
+from sentry.testutils.silo import control_silo_test
+
+
+@control_silo_test
+class ApiTokenDetailTest(APITestCase):
+    endpoint = "sentry-api-0-api-token-details"
+
+    def test_simple(self):
+        token = ApiToken.objects.create(user=self.user, name="token 1", scope_list=["event:read"])
+
+        self.login_as(self.user)
+        response = self.get_success_response(token.id, status_code=status.HTTP_200_OK)
+        assert response.content
+
+        res = response.data
+
+        assert res.get("id") == str(token.id)
+        assert res.get("name") == "token 1"
+        assert res.get("scopes") == ["event:read"]
+
+    def test_never_cache(self):
+        token = ApiToken.objects.create(user=self.user, name="token 1")
+
+        self.login_as(self.user)
+        response = self.get_success_response(token.id, status_code=status.HTTP_200_OK)
+        assert (
+            response.get("cache-control")
+            == "max-age=0, no-cache, no-store, must-revalidate, private"
+        )
+
+    def test_not_exists(self):
+        self.login_as(self.user)
+        self.get_error_response(-1, status_code=status.HTTP_404_NOT_FOUND)
+
+    def test_no_auth(self):
+        token = ApiToken.objects.create(user=self.user, name="token 1")
+        self.get_error_response(token.id, status_code=status.HTTP_401_UNAUTHORIZED)
+
+
+@control_silo_test
+class ApiTokenEditTest(APITestCase):
+    endpoint = "sentry-api-0-api-token-details"
+    method = "PUT"
+
+    def test_simple(self):
+        token = ApiToken.objects.create(user=self.user, name="token 1")
+        payload = {"name": "new token"}
+
+        self.login_as(self.user)
+        self.get_success_response(token.id, status_code=status.HTTP_204_NO_CONTENT, **payload)
+
+        tokenNew = ApiToken.objects.get(user=self.user)
+        assert tokenNew.name == "new token"
+        assert tokenNew.get_scopes() == token.get_scopes()
+
+    def test_never_cache(self):
+        token = ApiToken.objects.create(user=self.user, name="token 1")
+        payload = {"name": "new token"}
+
+        self.login_as(self.user)
+        response = self.get_success_response(
+            token.id, status_code=status.HTTP_204_NO_CONTENT, **payload
+        )
+        assert (
+            response.get("cache-control")
+            == "max-age=0, no-cache, no-store, must-revalidate, private"
+        )
+
+    def test_delete_name(self):
```

Comment:
to not be confused with the `DELETE` method, lets call this `remove_name` instead